src/main/java/edu/cornell/kfs/coa/batch/service/impl/CopyLegacyAccountAttachmentsServiceImpl.java

```diff
@@ -0,0 +1,232 @@
+                            maxFailuresAllowed);
+                    return false;
+                }
+            }
+        }
+
+        LOG.info("====================");
+        LOG.info("copyLegacyAccountAttachmentsToKfs, Successfully copied {} legacy account attachments into KFS",
+                numCopiedAttachments);
+        LOG.info("copyLegacyAccountAttachmentsToKfs, {} legacy account attachments failed copying into KFS",
+                numCopyFailures);
+        LOG.info("====================");
+
+        return true;
+    }
+
+    @Transactional(propagation = Propagation.REQUIRES_NEW)
+    @Override
+    public boolean copyLegacyAccountAttachmentToKfs(final LegacyAccountAttachment legacyAccountAttachment) {
+        try {
+            LOG.debug("copyLegacyAccountAttachmentToKfs, Copying attachment: {}", legacyAccountAttachment);
+            final Account account = getKfsAccountForAttachment(legacyAccountAttachment);
+            final Attachment attachment = downloadAttachment(legacyAccountAttachment, account);
+            saveAttachmentToKfs(legacyAccountAttachment, attachment, account);
+            LOG.info("copyLegacyAccountAttachmentToKfs, Copied attachment with ID '{}' targeting account '{}-{}'",
+                    legacyAccountAttachment.getId(), legacyAccountAttachment.getKfsChartCode(),
+                    legacyAccountAttachment.getKfsAccountNumber());
+            copyLegacyAccountAttachmentsDao.markLegacyAccountAttachmentAsCopied(legacyAccountAttachment);
+            return true;
+        } catch (final Exception e) {
+            LOG.error("copyLegacyAccountAttachmentToKfs, Failed to copy attachment: {}", legacyAccountAttachment, e);
+            copyLegacyAccountAttachmentsDao.recordCopyingErrorForLegacyAccountAttachment(
+                    legacyAccountAttachment, e.getMessage());
+            return false;
+        }
+    }
+
+    private Account getKfsAccountForAttachment(final LegacyAccountAttachment legacyAccountAttachment) {
+        final Account account = accountService.getByPrimaryId(legacyAccountAttachment.getKfsChartCode(),
+                legacyAccountAttachment.getKfsAccountNumber());
+        if (ObjectUtils.isNull(account)) {
+            throw new RuntimeException("Account '" + legacyAccountAttachment.getKfsChartCode()
+                    + "-" + legacyAccountAttachment.getKfsAccountNumber() + "' does not exist");
+        }
+        return account;
+    }
+
+    private Attachment downloadAttachment(final LegacyAccountAttachment legacyAccountAttachment,
+            final Account account) throws URISyntaxException {
+        final AccountingXmlDocumentBackupLink backupLink = new AccountingXmlDocumentBackupLink();
+        backupLink.setCredentialGroupCode(CuCoaBatchConstants.DFA_ATTACHMENTS_GROUP_CODE);
+        backupLink.setFileName(legacyAccountAttachment.getFileName());
+        backupLink.setDescription(legacyAccountAttachment.getFileDescription());
```

Comment:
Should this be the formatted note text discussed on the jira?

Reply:
The attachment-downloading service doesn't seem to be actively using this description, so I don't think it matters much in this case. Nevertheless, I could look into restructuring this to use the intended note text.